python/src/graphql.rs

```diff
@@ -380,24 +411,39 @@ impl PyRaphtoryServer {
                 })
         });
 
-        Ok(PyRunningRaphtoryServer::new(join_handle, sender, port))
+        let mut server = PyRunningRaphtoryServer::new(join_handle, sender, port);
+        if let Some(server_handler) = &server.server_handler {
+            match PyRunningRaphtoryServer::wait_for_server_online(
+                &server_handler.client.url,
+                timeout_in_milliseconds,
+            ) {
+                Ok(_) => return Ok(server),
+                Err(e) => {
+                    PyRunningRaphtoryServer::stop_server(&mut server, py)?;
+                    Err(e)
+                }
+            }
+        } else {
+            Err(PyException::new_err("Failed to start server"))
+        }
     }
 
     /// Run the server until completion.
     ///
     /// Arguments:
     ///   * `port`: the port to use (defaults to 1736).
-    #[pyo3(signature = (port = 1736, log_level="INFO".to_string(),enable_tracing=false,enable_auth=false))]
+    #[pyo3(
+        signature = (port = 1736, enable_tracing = false, enable_auth = false)
+    )]
```

Comment:
Can the server be stopped with `ctrl-c`? If not, can you make an issue to look into enabling that?